lib/srv/desktop/tdp/proto.go

```diff
@@ -1424,14 +1431,17 @@ func encodeString(w io.Writer, s string) error {
 	return nil
 }
 
+const stringMaxLenError = "TDP string length exceeds allowable limit"
+
 func decodeString(r io.Reader, maxLen uint32) (string, error) {
 	var length uint32
 	if err := binary.Read(r, binary.BigEndian, &length); err != nil {
 		return "", trace.Wrap(err)
 	}
 
 	if length > maxLen {
-		return "", trace.BadParameter("TDP string length exceeds allowable limit")
+		io.CopyN(io.Discard, r, int64(length))
```

Comment:
So it's clear that it's intentional
```suggestion
		_, _ = io.CopyN(io.Discard, r, int64(length))
```